numpy/core/multiarray.py

```diff
@@ -1,1713 +1,9 @@
-"""
-Create the numpy.core.multiarray namespace for backward compatibility. In v1.16
-the multiarray and umath c-extension modules were merged into a single
-_multiarray_umath extension module. So we replicate the old namespace
-by importing from the extension module.
+from numpy._core import multiarray
+from ._utils import _raise_warning
 
-"""
+_raise_warning("*")
```

Comment:
The warning I get from this is not great:

```
DeprecationWarning: `numpy.core` has been made officially private and renamed to `numpy._core`. Accessing `_core` directly is discouraged, as members of `_core` can change without any warning at any time. You should use a public module instead that exports the attribute in question. If you still would like to access an attribute from it, please use `np._core.*`.
```

Note that the suggestion is wrong (it should point to np.core.multiarray, not np.core). But also it's not giving me the name of the symbol causing this issue, which it could.

Can we replace this with a module-level __getattr__ please? Also customize the warning for submodules of `np.core`.

Reply:
Based on our discussions, file-level warnings have been removed, so file-wise imports, such as:
```python
from numpy.core.multiarray import _reconstruct
```
raise no warning (allowing undisturbed loading of 1.x pickle files).

The warning is still present when accessing via `np.core.*`. Therefore:
```python
import numpy as np
np.core.multiarray
```
gives:
```
<ipython-input-1-725e8a67566f>:1: DeprecationWarning: `numpy.core` has been made officially private and renamed to `numpy._core`. Accessing `_core` directly is discouraged, as members of `_core` can change without any warning at any time. You should use a public module instead that exports the attribute in question. If you still would like to access an attribute from it, please use `np._core.multiarray`.
```